react-backend/client/src/cb-admin/pages/Visitor.js

```diff
@@ -0,0 +1,301 @@
+    return (
+      <PrintContainer>
+        <PrintHeaderRow>
+          {
+            state.cbLogoUrl
+              ? <CbLogo src={state.cbLogoUrl} alt="Business logo" />
+              : <CbLogo src={p2cLogo} alt="Power to change logo" />
+          }
+          <Heading flex={9}>{state.cbOrgName} QR code</Heading>
+        </PrintHeaderRow>
+        <QrCodePrint src={state.qrCodeUrl} alt="QR code" />
+        <Paragraph>
+          Please bring this QR code with you next time
+        </Paragraph>
+      </PrintContainer>
+    );
+  }
+
+  renderMain(state) {
+    const { errors, ...rest } = state;
+    const rows = [
+      { name: 'Visitor ID', value: rest.id },
+      { name: 'Name', value: rest.name },
+      { name: 'Gender', value: rest.gender },
+      { name: 'Year of birth', value: rest.yob },
+      { name: 'Email', value: rest.email },
+      { name: 'Phone number', value: rest.phoneNumber },
+      { name: 'Registration date', value: rest.registeredAt },
+    ];
+
+    return (
+      <Col>
+        <Nav>
+          <HyperLink to="/admin"> Back to dashboard </HyperLink>
+          <Heading flex={2}>Visitor profile</Heading>
+          <FlexItem />
+        </Nav>
+        <Row>
+          <FlexItem flex={7}>
+            <DetailsTable rows={rows} caption="Visitor details" />
+          </FlexItem>
+          <FlexItem flex={4}>
+            <QrBox
+              qrCodeUrl={rest.qrCodeUrl}
+              print={this.onClickPrint}
+              send={this.onClickResend}
+              hasSent={rest.hasResent}
+            />
+          </FlexItem>
+        </Row>
+        <FlexContainerRow flex={4}>
+          <Form onChange={this.onChange} onSubmit={this.onSubmit}>
+            <Paragraph>Edit user details</Paragraph>
+            <FlexItem flex={7}>
+              <LabelledInput
+                id="visitor-name"
+                label="Name"
+                name="name"
+                type="text"
+                error={errors.name}
+              />
+              <LabelledInput
+                id="visitor-email"
+                label="Email"
+                name="email"
+                type="email"
+                error={errors.email}
+              />
+              <Button type="submit">SAVE</Button>
+            </FlexItem>
+            <FlexItem flex={4}>
+              <LabelledInput
+                id="visitor-yob"
+                label="Year of birth"
+                name="yob"
+                type="text"
+                error={errors.yob}
+              />
+              <LabelledSelect
+                id="visitor-gender"
+                label="Gender"
+                name="gender"
+                options={genderOptions}
+                error={errors.gender}
+              />
+            </FlexItem>
+          </Form>
+        </FlexContainerRow>
+      </Col>
+    );
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        {
```

Comment:
Is this rendering both versions of the page, but PrinterFriendly is set to display none?

Reply:
Yes